RFC-005-SWAP-Bitcoin-basic.md

```diff
@@ -0,0 +1,248 @@
+
+### Refund
+
+To refund the HTLC, the funder MUST submit a transaction to the blockchain which spends the P2WSH output.
+The funder can use the following witness data to spend the output after the `expiry`:
+
+| Data             | Description                                                                                             |
+|:-----------------|:--------------------------------------------------------------------------------------------------------|
+| refund_signature | A valid SECP256k1 ECDSA DER encoded signature on the transaction with respect to the `refund_pubkey`    |
+| refund_pubkey    | The 33 byte SECP256k1 compressed public key that was hashed to produce the pubkeyhash `refund_identity` |
+| `00`             | A single byte used to activate the refund path in the `OP_IF`                                           |
+| contract_script  | The compiled contract (as generally required when redeeming from a P2WSH output)                        |
+
+To be notified of the refund event, both parties MAY watch the blockchain for transactions that spend from the output and check that the witness data is in the above form.
+
+## Registry extension
+
+This RFC extends the [registry](./registry.md#identities) with an identity definition for the Bitcoin ledger:
+
+| Ledger  | Identity Name | JSON Encoding            | Description                                                                           |
+|:--------|:--------------|:-------------------------|---------------------------------------------------------------------------------------|
+| Bitcoin | `pubkeyhash`  | `hex-encoded-bytes (20)` | The result of applying SHA-256 and then RIPEMD-160 to a SECP256k1 compressed public key |
+
+# Examples
+
+## RFC003 SWAP REQUEST
+
+The following shows an [RFC003](RFC-003-SWAP-basic.md) SWAP REQUEST where the `alpha_ledger` is Bitcoin, the `alpha_asset` is 1 Bitcoin (with `...` being used where the value is only relevant for the `beta_ledger`).
+
+``` json
+{
+  "type": "SWAP",
+  "headers": {
+    "alpha_ledger": {
+      "value": "bitcoin",
+      "parameters": { "network": "mainnet" }
+    },
+    "beta_ledger": {...},
+    "alpha_asset": {
+      "value": "bitcoin",
+      "parameters": { "quantity": "100000000" }
+    },
+    "beta_asset": {...},
+    "protocol": {
+        "value" : "comit-rfc-003",
+        "parameters" : { "hash_function" : "SHA-256" }
+    }
+  },
+  "body": {
+    "aplha_ledger_refund_identity": "1925a274ac004373bb5429553bdb55c40e57b124",
+    "alpha_expiry": 1552263040,
+    "secret_hash" : "51a488e06e9c69c555b8ad5e2c4629bb3135b96accd1f23451af75e06d3aee9c",
+    "beta_ledger_redeem_identity" : "...",
+    "beta_expiry" : ...
+  },
+}
+```
+
+Note, the secret for the `secret_hash` is `51a488e06e9c69c555b8ad5e2c4629bb3135b96accd1f23451af75e06d3aee9c`.
```

Comment:
:x: How can this be true if both values are identical?

Reply:
Well spotted!